.github/workflows/lint.yml

```diff
@@ -0,0 +1,26 @@
+---
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  lint:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # this is to enable gitlint to check all PR commit messages
+          submodules: 'true'
+
+      - name: Install Dependencies
+        run: sudo snap install pre-commit --classic
```

Comment:
Flags generally should come before arguments - some commands will break if you put them the other way around. So this should probably be `sudo snap install --classic pre-commit`